packages/blade/src/components/Button/IconButton/StyledIconButton.native.tsx

```diff
@@ -0,0 +1,49 @@
+import type { ReactElement } from 'react';
+import { useState } from 'react';
+import styled from 'styled-components/native';
+
+import type { StyledIconButtonProps } from './StyledIconButton';
+import { makeAccessible } from '~utils';
+import type { ColorContrastTypes } from '~tokens/theme/theme';
+
+type State = 'active' | 'default';
+type IconColorToken = `feedback.neutral.action.icon.link.${State}.${ColorContrastTypes}Contrast`;
+
+type StyledPressableProps = {
+  contrast: ColorContrastTypes;
+};
+
+const StyledPressable = styled.Pressable<StyledPressableProps>({
+  alignSelf: 'center', // ensure button only takes needed width
+});
+
+const StyledIconButton = ({
+  icon: Icon,
+  onClick,
+  size,
+  contrast,
+  accessibilityLabel,
+}: StyledIconButtonProps): ReactElement => {
+  const [isPressed, setIsPressed] = useState(false);
+  const getIconColorToken = (): IconColorToken => {
+    const contrastType = contrast === 'high' ? 'highContrast' : 'lowContrast';
+    const state = isPressed ? 'active' : 'default';
+
+    return `feedback.neutral.action.icon.link.${state}.${contrastType}`;
+  };
```

Comment:
We should extract this out of the component.

Reply:
Did that first but later inlined it. [Ref](https://github.com/razorpay/blade/pull/613#discussion_r944114607)

Reply:
Fine I guess. Just it will get recreated on each render.